src/clarity-angular/utils/focus-trap/focus-trap-linker.service.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Injectable} from "@angular/core";
+import {FocusTrapDirective} from "./focus-trap.directive";
+
+@Injectable()
+export class FocusTrapLinker {
+    currentFocusTrap: FocusTrapDirective;
+    previousFocusTrap: FocusTrapDirective;
```

Comment:
What if you have 3 modals? You support 2, but draw the line at 3? That seems arbitrary.
Please change this to a queue so that if we support nested modals, we support any number of them.

Reply:
No, it can support any number of modals. FocusTrapLinker is merely playing the role of a linker; it's not responsible for the chain of references. The chain of references is saved in the `this.focusTrapLinker.previousFocusTrap`. So no limit on the number of modals.

Reply:
Yes, there is a limit, you don't have a reference to the third modal anymore. Here is a step-by-step example, using your code (there are several issues in this scenario): let's say you have 3 modals A, B and C, and you open them one by one without closing the previous one.
1. Open modal A
```typescript
currentFocusTrap = A
previousFocusTrap = undefined
```
2. Open modal B
```typescript
currentFocusTrap = B
previousFocusTrap = A
```
3. Open modal C
```typescript
currentFocusTrap = C
previousFocusTrap = B
```
4. Close modal C
```typescript
currentFocusTrap = B
previousFocusTrap = B
```
(We already see something is wrong.)
5. Close modal B
```typescript
currentFocusTrap = B
previousFocusTrap = B
```
No focus trap on A, focus trap on B which has been destroyed, ... It's all broken.

Reply:
Oh lol, I just forgot to set `prevFocusTrap` in the `focus-trap.directive.ts`. If I set a value to it, I think it can support any number of modals. But, I will see and test it more than 3 nested modals.